### PR TITLE
DOC: add dropdown

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

This won't work because the json file needs to be hard coded versus relatively coded, and I'm also not sure we pull pydata-sphinx-theme 0.8   However, when we do, and when this gets merged into stable, it will allow a drop-down menu of version info.  See https://pydata-sphinx-theme.readthedocs.io/en/latest/

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by jklymak** [comment]
I think this will work fine on devdocs once merged.  Note there is another step here for the release manager...

File: doc/_static/switcher.json
Comment (timhoffm) [inline doc/_static/switcher.json:8]:
```suggestion
        "version": "devdocs"
```

File: doc/_static/switcher.json
Comment (timhoffm) [inline doc/_static/switcher.json:3]:
```suggestion
        "name": "3.5.1 (latest)",
```
I assume it looks better in the dropdown if all entries start with version numbers. I also think that "latest" is a better description than "stable".

File: doc/_static/switcher.json
Comment (timhoffm) [inline doc/_static/switcher.json:7]:
```suggestion
        "name": "3.6 (devel)",
```

**Review by timhoffm — COMMENTED** [review]
Please add a description what needs to be done by the release manager 
 to the release guide.

File: doc/_static/switcher.json
Comment (jklymak) [reply doc/_static/switcher.json:3]: Well, except the directory is `stable`, so we should probably follow that?

File: doc/_static/switcher.json
Comment (jklymak) [reply doc/_static/switcher.json:8]: Ooops

File: doc/_static/switcher.json
Comment (jklymak) [reply doc/_static/switcher.json]: .. and I changed this to `"3.6 (devdocs)"` to be consistent with my argument above ;-)  However, I don't feel strongly if we should rename these.  

I _do_ hope we can merge this, just to see if it works on the devdocs.  We can always fine-tune it later.

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Which milestone do you want?

**Review by dstansby — COMMENTED** [review]
Does the theme doc requirement need pinning to >=0.8 now?

**Comment by dstansby** [comment]
Also this doesn't seem to be working on the built docs? https://66341-1385122-gh.circle-artifacts.com/0/doc/build/html/index.html

**Comment by jklymak** [comment]
It doesn't work because it requires the json to be hard coded on the web somewhere.  Which merging this pr will do ;-).  

We could try a backport after we are sure this works.

File: doc/_static/switcher.json
Comment (dstansby) [inline doc/_static/switcher.json:8]: Do we actually want this showing up in the user-facing version options? Perhaps it should go at the bottom?

File: doc/_static/switcher.json
Comment (dstansby) [inline doc/_static/switcher.json:15]: Thoughts on just having a `3.4` option that points to the latest `3.4.x` release instead of listing all the 3.4.x releases? Or just having a `3.4.3` option?

Comment (jklymak) [reply doc/_static/switcher.json:15]: Thats what I did for 2.4 and older, but not here.

File: doc/_static/switcher.json
Comment (jklymak) [reply doc/_static/switcher.json:8]: We link it prominently on the front page, so I think it should be at the top

File: doc/_static/switcher.json
Comment (jklymak) [reply doc/_static/switcher.json:15]: Just to be clear, I'm not even sure this works.  I vote we merge it, and then fix any stylistics choices post-merge when we can see what it is like.

**Review by dstansby — APPROVED** [review]

**Pull request merged**
